### ISSUE TITLE
Range is inclusive on both ends

### ISSUE THREAD

**Issue by Maltysen** [cheddar-lang/Cheddar#43]
### Current Behavior

```
cheddar> 4:4
 [4]
cheddar> 2:10
 [2, 3, 4, 5, 6, 7, 8, 9, 10]
```

### Desired Behavior

`2:10` giving `[2,3,4,5,6,7,8,9]`
and `4:4` giving `[]`

because now, `0:n` has a length of `n+1`, not `n`

**Comment by vihanb** [comment]
well what should be the desired behavior?

**Comment by Maltysen** [comment]
`2:10` giving `[2,3,4,5,6,7,8,9]`

**Comment by Maltysen** [comment]
because now, `0:n` has a length of `n+1`, not `n`

**Comment by ConorOBrien-Foxx** [comment]
I don't see why this is a problem.

**Comment by Maltysen** [comment]
@ConorOBrien-Foxx array indexing, (I'm guessing, based on js) will be 0-indexed

**Comment by ConorOBrien-Foxx** [comment]
Why does that follow ?

**Comment by Maltysen** [comment]
@ConorOBrien-Foxx  what happens when you want to loop through an array of len `n`? the indices of that array are `0..n-1`

**Comment by ConorOBrien-Foxx** [comment]
Okay? I still don't see your point. In a range m:n, n is often not the length of the range.

**Comment by Maltysen** [comment]
@ConorOBrien-Foxx but you will be indexing `0:n` for that array

**Comment by ConorOBrien-Foxx** [comment]
Could you maybe rephrase? Perhaps I'll understand then.

**Comment by Maltysen** [comment]
@ConorOBrien-Foxx  http://stackoverflow.com/questions/4504662/why-does-rangestart-end-not-include-end

**Comment by ConorOBrien-Foxx** [comment]
Oh, I see. It's customary.

**Comment by ConorOBrien-Foxx** [comment]
Fixed by making monadic range do this.
